tests/proxy/replicated_map/replicated_map_test.go

```diff
@@ -0,0 +1,345 @@
+	res, err := rmp.Get(testKey)
+	AssertNilf(t, err, res, "PutWithTtl() failed")
+}
+
+func TestReplicatedMapProxy_PutAll(t *testing.T) {
+	defer rmp.Clear()
+	testMap := make(map[interface{}]interface{})
+	for i := 0; i < 10; i++ {
+		testMap["testingKey"+strconv.Itoa(i)] = "testingValue" + strconv.Itoa(i)
+	}
+	err := rmp.PutAll(testMap)
+	AssertErrorNil(t, err)
+	entryList, err := rmp.EntrySet()
+	AssertErrorNil(t, err)
+	for _, pair := range entryList {
+		key := pair.Key()
+		value := pair.Value()
+		expectedValue, found := testMap[key]
+		if !found || expectedValue != value {
+			t.Errorf("PutAll() failed")
+		}
+	}
+}
+
+func TestReplicatedMapProxy_Get(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	_, err := rmp.Put(testKey, testValue)
+	res, err := rmp.Get(testKey)
+	AssertEqualf(t, err, res, testValue, "Get() failed")
+}
+
+func TestReplicatedMapProxy_ContainsKey(t *testing.T) {
+	defer rmp.Clear()
+	for i := 1; i <= 10; i++ {
+		rmp.Put("testKey"+strconv.Itoa(i), "testValue"+strconv.Itoa(i))
+	}
+	res, err := rmp.ContainsKey("testKey5")
+	AssertEqualf(t, err, res, true, "ContainsKey() failed")
+}
+
+func TestReplicatedMapProxy_ContainsValue(t *testing.T) {
+	defer rmp.Clear()
+	for i := 1; i <= 10; i++ {
+		rmp.Put("testKey"+strconv.Itoa(i), "testValue"+strconv.Itoa(i))
+	}
+	res, err := rmp.ContainsValue("testValue5")
+	AssertEqualf(t, err, res, true, "ContainsValue() failed")
+}
+
+func TestReplicatedMapProxy_Clear(t *testing.T) {
+	for i := 1; i <= 10; i++ {
+		rmp.Put("testKey"+strconv.Itoa(i), "testValue"+strconv.Itoa(i))
+	}
+	err := rmp.Clear()
+	size, _ := rmp.Size()
+	AssertEqualf(t, err, size, int32(0), "Clear() failed")
+}
+
+func TestReplicatedMapProxy_Remove(t *testing.T) {
+	defer rmp.Clear()
+	for i := 1; i <= 10; i++ {
+		rmp.Put("testKey"+strconv.Itoa(i), "testValue"+strconv.Itoa(i))
+	}
+	value, err := rmp.Remove("testKey5")
+	AssertEqualf(t, err, value, "testValue5", "Remove() failed")
+	size, _ := rmp.Size()
+	AssertEqualf(t, err, size, int32(9), "Remove() failed")
+}
+
+func TestReplicatedMapProxy_IsEmpty(t *testing.T) {
+	for i := 1; i <= 10; i++ {
+		rmp.Put("testKey"+strconv.Itoa(i), "testValue"+strconv.Itoa(i))
+	}
+	empty, err := rmp.IsEmpty()
+	AssertEqualf(t, err, empty, false, "IsEmpty() failed")
+	rmp.Clear()
+	empty, err = rmp.IsEmpty()
+	AssertEqualf(t, err, empty, true, "IsEmpty() failed")
+}
+
+func TestReplicatedMapProxy_Size(t *testing.T) {
+	defer rmp.Clear()
+	for i := 1; i <= 10; i++ {
+		rmp.Put("testKey"+strconv.Itoa(i), "testValue"+strconv.Itoa(i))
+	}
+	size, err := rmp.Size()
+	AssertEqualf(t, err, size, int32(10), "Size() failed")
+}
+
+func TestReplicatedMapProxy_Values(t *testing.T) {
+	defer rmp.Clear()
+	var expecteds = make([]string, 10)
+	var ret = make([]string, 10)
+	for i := 0; i < 10; i++ {
+		rmp.Put(strconv.Itoa(i), strconv.Itoa(i))
+		expecteds[i] = strconv.Itoa(i)
+	}
+	values, err := rmp.Values()
+	for j := 0; j < 10; j++ {
```

Comment:
what if `values` was broken so that it returned more elements than 10. This test would still pass. It is better if you write a utility method for this kind of comparisons and use it in these kind of tests. Something like `AssertIterableEqual`.